catalogs/bundled-catalog/build.gradle.kts

```diff
@@ -85,6 +85,7 @@ tasks.withType<ShadowJar>(ShadowJar::class.java) {
   exclude("NOTICE")
 
   minimize()
+  relocate("com.datastrato.gravitino.catalog.lakehouse.iceberg", "com.datastrato.gravitino.shaded.bundled.catalog.lakehouse.iceberg")
```

Comment:
I'm not very sure why only to relocate the package of Iceberg catalog, are there any similar issues for Hive and Hadoop catalogs?

Reply:
no, it had problem when the classes referred by `XXXPropertiesMetadata` is moved from the module

Reply:
I see. Relocating Iceberg catalog-related code is not so elegant but I can't come up with a better choice.

Reply:
I think this is not what bundled-catalog originally proposed to do, it is weird to shade Iceberg catalog package in bundled-catalog.

With this, there may have two Gravitino Iceberg package, one shaded and one not, will this lead to problem easily?

Reply:
bundle catalog is used to provide the constants shared by connectors and Gravitino catalogs. The current implementation used the constant from `XXXPropertiesMetadata` and introduced the referred classes in `bundle-catalog`,  seems not proper, How about defining constant in `bundle catalog` and `XXXPropertiesMetadata` rely on the constants?

Reply:
Yeah, that's better than the current solution.

Reply:
Propose #4122 to use constants defined in `bundled catalog`, @jerryshao @yuqi1129 please help to review, I'll close this PR